lib/spack/spack/spec.py

```diff
@@ -716,9 +724,10 @@ def canonical(self):
 
 class FlagMap(lang.HashableMap):
 
-    def __init__(self, spec):
+    def __init__(self, spec, propagate=False):
         super(FlagMap, self).__init__()
         self.spec = spec
+        self.propagate = propagate
```

Comment:
This looks like it sets propagate for all compiler flags, rather than individually. What about `spack spec tcl cflags==-g cxxflags=-O3`? I think we need a test to confirm that propagates the appropriate one but not the other.

Reply:
You're right. Neither compiler flag propagated. Thanks for catching that.